cmd/access/node_builder/access_node_builder.go

```diff
@@ -680,39 +624,388 @@ func unstakedNetworkMsgValidators(log zerolog.Logger, idProvider id.IdentityProv
+		if err != nil {
+			return nil, fmt.Errorf("could not register networking receive cache metric: %w", err)
+		}
+
+		net, err := builder.initNetwork(builder.Me, builder.PublicNetworkConfig.Metrics, middleware, top, receiveCache)
 		if err != nil {
 			return nil, err
 		}
 
-		// create the identity of the peer by setting only the relevant fields
-		ids[i] = &flow.Identity{
-			NodeID:        flow.ZeroID, // the NodeID is the hash of the staking key and for the unstaked network it does not apply
-			Address:       address,
-			Role:          flow.RoleAccess, // the upstream node has to be an access node
-			NetworkPubKey: networkKey,
+		builder.AccessNodeConfig.PublicNetworkConfig.Network = net
+
+		node.Logger.Info().Msgf("network will run on address: %s", builder.PublicNetworkConfig.BindAddress)
+		return net, nil
+	})
+}
+
+// initLibP2PFactory creates the LibP2P factory function for the given node ID and network key.
+// The factory function is later passed into the initMiddleware function to eventually instantiate the p2p.LibP2PNode instance
+// The LibP2P host is created with the following options:
+// 		DHT as server
+// 		The address from the node config or the specified bind address as the listen address
+// 		The passed in private key as the libp2p key
+//		No connection gater
+// 		Default Flow libp2p pubsub options
+func (builder *FlowAccessNodeBuilder) initLibP2PFactory(networkKey crypto.PrivateKey) p2p.LibP2PFactoryFunc {
+	return func(ctx context.Context) (*p2p.Node, error) {
+		connManager := p2p.NewConnManager(builder.Logger, builder.PublicNetworkConfig.Metrics)
+
+		libp2pNode, err := p2p.NewNodeBuilder(builder.Logger, builder.PublicNetworkConfig.BindAddress, networkKey, builder.SporkID).
+			SetBasicResolver(builder.Resolver).
+			SetSubscriptionFilter(
+				p2p.NewRoleBasedFilter(
+					flow.RoleAccess, builder.IdentityProvider,
+				),
+			).
+			SetConnectionManager(connManager).
+			SetRoutingSystem(func(ctx context.Context, h host.Host) (routing.Routing, error) {
+				return p2p.NewDHT(ctx, h, unicast.FlowPublicDHTProtocolID(builder.SporkID), p2p.AsServer())
+			}).
+			SetPubSub(pubsub.NewGossipSub).
+			Build(ctx)
+
+		if err != nil {
+			return nil, fmt.Errorf("could not build libp2p node for staked access node: %w", err)
 		}
+
+		builder.LibP2PNode = libp2pNode
+
+		return builder.LibP2PNode, nil
 	}
-	return ids, nil
+}
+
+// initMiddleware creates the network.Middleware implementation with the libp2p factory function, metrics, peer update
+// interval, and validators. The network.Middleware is then passed into the initNetwork function.
+func (builder *FlowAccessNodeBuilder) initMiddleware(nodeID flow.Identifier,
+	networkMetrics module.NetworkMetrics,
+	factoryFunc p2p.LibP2PFactoryFunc,
+	validators ...network.MessageValidator) network.Middleware {
+
+	// disable connection pruning for the staked AN which supports the unstaked AN
```

Comment:
update this comment

Reply:
Addressed: https://github.com/GetElastech/flow-go/commit/a33fc58d87da30525dbfc69bb1689ad931bb0636